sql/CMakeLists.txt

```diff
@@ -180,6 +180,13 @@ IF(INTERFACE_LIBS)
   "${INTERFACE_LIBS}")
 ENDIF()
 
+IF(WITH_SYSTEMD)
+# Used for notify systemd's feature
+IF(HAVE_SYSTEMD_SD_DAEMON_H)
+  TARGET_LINK_LIBRARIES(mysqld systemd-daemon)
+ENDIF()
```

Comment:
* I've already commented on "all or nothing" approach
* why do you need that at all, you add ${LIBSYSTEMD} to the list of libraries above